interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * The volcano-plot module provides a directive and controller for embedding
+ * volcano plots in other views. Requires samples to be grouped in the
+ * adage.analyze.sampleBin in order to function properly -- if no samples are
+ * chosen or placed into groups, an error will be displayed.
+ */
+angular.module('adage.volcano-plot', [
+  'ngVega',
+  'greenelab.stats',
+  'adage.analyze.sampleBin',
+  'adage.volcano-plot-vgspec'
+])
+
+.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec', 'SampleBin',
+  function VolcanoPlotCtrl($scope, VolcanoPlotSpec, SampleBin) {
+    // retrieve the nodeInfo map (for node names) via the SampleBin service
+    // $scope.nodeInfo = SampleBin.getNodeInfo(); // TBD
+    $scope.nodeInfo = [
```

Comment:
Yes, this is test data. I started writing the fetch for the actual `nodeInfo` data in line 17, above, but decided to get the vega part working first. I haven't circled back to remove it because I'm still getting that part to work.